pkg/util/clusteroperator/status_manager.go

```diff
@@ -0,0 +1,184 @@
+package clusteroperator
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StatusManager coordinates changes to ClusterOperator.Status
+type StatusManager struct {
+	client  client.Client
+	name    string
+	version string
+
+	configFailure bool
+}
+
+func NewStatusManager(client client.Client, name, version string) *StatusManager {
+	return &StatusManager{client: client, name: name, version: version}
+}
+
+// Set updates the ClusterOperator.Status with the provided conditions
+func (status *StatusManager) Set(conditions ...*configv1.ClusterOperatorStatusCondition) error {
+	co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: status.name}}
+	err := status.client.Get(context.TODO(), types.NamespacedName{Name: status.name}, co)
+	isNotFound := errors.IsNotFound(err)
+	if err != nil && !isNotFound {
+		log.Printf("failed to get clusteroperator %s: %v", status.name, err)
+		return fmt.Errorf("failed to get clusteroperator %s: %v", status.name, err)
+	}
+
+	oldConditions := co.Status.Conditions
+	for _, condition := range conditions {
+		co.Status.Conditions = SetStatusCondition(co.Status.Conditions, condition)
+	}
+	co.Status.Version = status.version
```

Comment:
It looks like the latest API changes (https://github.com/openshift/api/pull/176) changes how this works.

In the interest of getting this out the door, let's just drop this line, revendor api, and merge the PR, then figure out the operand stuff afterwards.